src/AsmResolver.DotNet/Signatures/GenericContext.cs

```diff
@@ -18,39 +19,45 @@ public GenericContext(IGenericArgumentsProvider? type, IGenericArgumentsProvider
             Type = type;
             Method = method;
         }
-        
+
         /// <summary>
-        /// Gets the object responsible for providing type arguments defined by the current generic type instantiation. 
+        /// Gets the object responsible for providing type arguments defined by the current generic type instantiation.
         /// </summary>
         public IGenericArgumentsProvider? Type
         {
             get;
         }
 
         /// <summary>
-        /// Gets the object responsible for providing type arguments defined by the current generic method instantiation. 
+        /// Gets the object responsible for providing type arguments defined by the current generic method instantiation.
         /// </summary>
         public IGenericArgumentsProvider? Method
         {
             get;
         }
 
         /// <summary>
-        /// Enters a new generic context with a new type providing type arguments. 
+        /// Returns true if both Type and Method providers are null
+        /// </summary>
+        public bool IsEmpty => Type is null && Method is null;
+
+        /// <summary>
+        /// Enters a new generic context with a new type providing type arguments.
         /// </summary>
         /// <param name="type">The new type providing the type arguments.</param>
         /// <returns>The new generic context.</returns>
         public GenericContext WithType(IGenericArgumentsProvider type) => new GenericContext(type, Method);
-        
+
         /// <summary>
-        /// Enters a new generic context with a new method providing type arguments. 
+        /// Enters a new generic context with a new method providing type arguments.
         /// </summary>
         /// <param name="method">The new method providing the type arguments.</param>
         /// <returns>The new generic context.</returns>
         public GenericContext WithMethod(IGenericArgumentsProvider method) => new GenericContext(Type, method);
 
         /// <summary>
         /// Resolves a type parameter to a type argument, based on the current generic context.
+        /// If generic argument provider is not initialised for type parameter, than it will return itself.
```

Comment:
This sentence is not really clear.
- What is meant by _if it is not initialized for type parameter_? 
- The word _itself_ can now refer to the generic argument provider or to the type parameter.

I think you mean to say something in the lines of: 

> If a type parameter within the signature references a parameter that is not captured by the context (i.e. the corresponding generic argument provider is set to `null`), then this type parameter will not be substituted.


We might consider moving this to the `<remarks>` section as well.